packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -1141,8 +1146,23 @@ class DoNothingIntent extends Intent {
 ///  - [DoNothingIntent], which is an intent that can be bound to a keystroke in
 ///    a [Shortcuts] widget to do nothing.
 class DoNothingAction extends Action<Intent> {
+  /// Creates a const [DoNothingAction].
+  ///
+  /// The optional [shouldHandleKey] argument defaults to true.
+  DoNothingAction({this.shouldHandleKey = true});
+
   @override
   void invoke(Intent intent) {}
+
+  /// Indicates whether this action should mark any key event mapped to this
+  /// action as being "handled", or not.
+  ///
+  /// If the key is not handled, it will be passed back to the engine, and
+  /// continue to be processed.
+  ///
+  /// If the key is handled, then no further processing will happen to that key
+  /// event.
+  final bool shouldHandleKey;
```

Comment:
Who is reading and acting on this property? Maybe I missed it, but I couldn't find this get used in the PR?

Reply:
It's magic, apparently. :-)

Actually, I tried to add the missing code that handles this, but I realized that now that we can't stop propagation, this flag doesn't really do anything useful, since if it doesn't handle it, it will just keep going up the tree to find another `Shortcuts` widget and give it to that widget, effectively nullifying the "do nothing" aspect of the `DoNothingAction`.

If we're going to handle it here, then there needs to be a way to return a trinary value from `handleKeypress` (e.g. `bool?`) so that the event propagation code in `FocusManager` can know to stop propagation. In that case, though the signature of `FocusOnKeyCallback` would change, which is a pretty big breaking change (would break anyone listening to keys).

I suppose I could add another kind of callback to `FocusNode` that allows a trinary return type, and make it so that only one of `onKey` or that new callback can be set. Feels kind of messy, though.

Actually, if I did it, I'd have it return an enum. not a `bool?` in case more cases crop up in the future.

Reply:
+1 to using a enum instead of bool?

Reply:
+1 to using a enum instead of bool?